presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1061,6 +1061,9 @@ private static Table buildTableObject(
 
         if (external) {
             tableParameters.put("EXTERNAL", "TRUE");
+            // collecting stats (like number of files) at table creation time can cause the the DDL statement to time out
+            // when the table location contains a large number of files
+            tableParameters.put("DO_NOT_UPDATE_STATS", "true");
```

Comment:
Is it suitable to external tables only?

(it's likely, but please mention this in a comment or link to apache/hive repo, to the place where this is being `if`-d on)

Reply:
Also, do we need to fill `tableParameters`, or only the `EnvironmentContext`?

Reply:
> Is it suitable to external tables only?
Hive doesn't seem to make a distinction, but this you cannot create a managed table that points to a location with data.

> Also, do we need to fill tableParameters, or only the EnvironmentContext?
For version 3.x, providing the table parameter to Hive is sufficient.  Hive will skip collecting stats and will remove the parameters (see https://github.com/apache/hive/blob/branch-3.1/standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java#L668-L684).  There is a comment in there that says "We want to avoid this property from being persistent".  In Presto, we use the table property as a way to communicate that the intent is to not collect stats.

Reply:
> Is it suitable to external tables only?

I tried to restrict to external table creation because managed tables cannot be created in a location that already contains data, so they're not affected by this problem.  Hive doesn't really make a distinction.

> Also, do we need to fill tableParameters, or only the EnvironmentContext?

I was initially using table parameters as a vehicle to inform `ThriftHiveMetastore` that it needed to make a different call.  But after our offline conversation where you pointed out that we needed to support Hive 1.x, I discovered that the context parameter is ignored in that version, however the table property is honored.